toolchain/check/handle_class.cpp

```diff
@@ -57,7 +57,7 @@ static auto MergeClassRedecl(Context& context, Parse::AnyClassDeclId node_id,
                              SemIR::ClassId prev_class_id,
                              SemIR::ImportIRId prev_import_ir_id) -> bool {
   auto& prev_class = context.classes().Get(prev_class_id);
-  SemIR::LocId prev_loc_id = prev_class.latest_decl_id();
+  auto prev_loc_id = SemIR::LocId(prev_class.latest_decl_id());
```

Comment:
I think https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/cpp_style_guide.md#syntax-and-formatting prefers parenthesized initialization for cases like this. The assignment I think is more about things like `int i = 3;` or `MyStruct foo = {...};`, where you're not explicitly calling the constructor on the RHS.

i.e.:
```suggestion
  SemIR::LocId prev_loc_id(prev_class.latest_decl_id());
```

I think there are a handful of similar spots in this PR.

Reply:
Ah.. yeah I guess so. I really don't like mixing `T x(...)` syntax into modern code, it's sometimes the vexxing parse and it reads so different than using `=` which dominates the code otherwise. But style guide is gonna style guide.

Reply:
Tob e sure, this is really about steering people away from the ambiguous cast behaviors [noted on #toolchain](https://discord.com/channels/655572317891461132/655578254970716160/1365056553073512608)